builtin/logical/transit/path_encrypt.go

```diff
@@ -127,6 +127,82 @@ to the min_encryption_version configured on the key.`,
 	}
 }
 
+// decodeBatchRequestItems is a fast path alternative to mapstructure.Decode to decode []BatchRequestItem.
+func decodeBatchRequestItems(src interface{}, dst *[]BatchRequestItem) error {
+	if src == nil || dst == nil {
+		return nil
+	}
+
+	items, ok := src.([]interface{})
+	if !ok {
+		return fmt.Errorf("source data must be an array or slice, got %T", src)
+	}
+
+	// Early return should happen before allocating the array if the batch is empty.
+	// However to comply with mapstructure output we must allocate an empty array.
+	sitems := len(items)
+	*dst = make([]BatchRequestItem, sitems)
```

Comment:
I consider it good practice to never yield a partial output when an error is returned.  The usual pattern is to write what you're planning to return into a local variable, which only gets returned (or in this case, written to the `dst` output parameter) on success.